apis/v1alpha2/httproute_types.go

```diff
@@ -663,7 +686,43 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
-// HTTPRequestRedirectFilter defines configuration for the RequestRedirect filter.
+// HTTPPathModifierType defines the type of path redirect.
+type HTTPPathModifierType string
+
+const (
+	// This type of modifier indicates that the complete path will be replaced
+	// by the path redirect value.
+	AbsoluteHTTPPathModifier HTTPPathModifierType = "Absolute"
+
+	// This type of modifier indicates that any prefix path matches will be
+	// replaced by the substitution value. For example, a path with a prefix
+	// match of "/foo" and a ReplacePrefixMatch substitution of "/bar" will have
+	// the "/foo" prefix replaced with "/bar" in matching requests.
+	PrefixMatchHTTPPathModifier HTTPPathModifierType = "ReplacePrefixMatch"
+)
+
+// HTTPPathModifier defines configuration for path modifiers.
+// <gateway:experimental>
+type HTTPPathModifier struct {
+	// Type defines the type of path modifier.
+	//
+	// <gateway:experimental>
+	// +kubebuilder:validation:Enum=Absolute;ReplacePrefixMatch
+	Type HTTPPathModifierType `json:"type"`
+
+	// Substitution defines the HTTP path value to substitute. An empty value
+	// ("") indicates that the portion of the path to be changed should be
+	// removed from the resulting path. For example, a request to "/foo/bar"
+	// with a prefix match of "/foo" would be modified to "/bar".
```

Comment:
I'm a bit hesitant about empty fields, especially string semantics. They are not usually the most intuitive, even more so in kube land.
Have we considered another "Type": "RemovePrefixMatch"?

Reply:
What if explicitly a NULL is the substitution value for the remove case?  Something explicit in the value as opposed to an additional type (too many types, while explicit to implement can cause usability issues and API friction)
Also, many JSON parsers and automation tools drop an empty value - thus preventing passing an empty value in a document through a tool chain, and YAML empty value = NULL

Reply:
@hbagdi We discussed this at the community meeting yesterday. I suggested something similar to what @brianehlert said above, essentially trying to ensure that users had to specify at least something here, even if that was just "". I tested the current CRDs out with the following YAML in a redirect filter:

```yaml
path:
  type: ReplacePrefixMatch
```

That resulted in the following error:
```
error validating data: ValidationError(HTTPRoute.spec.rules[0].filters[0].requestRedirect.path): missing required field "substitution" in io.k8s.networking.gateway.v1alpha2.HTTPRoute.spec.rules.filters.requestRedirect.path
```

That error could be avoided by specifying an empty string:

```yaml
path:
  type: ReplacePrefixMatch
  substitution: ""
```

I think that ends up being pretty clear. 

The other key point that came up in the community meeting was that "replace with nothing" is a fairly common way to represent "remove" in a wide variety of tooling. Although I agree that we need to be careful about inferring meaning from unspecified values, I think we're safely avoiding that by requiring user input. 

With all that said, my personal preference is to avoid an additional unique filter type here and instead move forward with the ability to replace a prefix match with nothing.